### PR TITLE
Rename all `*Parameters` to `*Config`

### PR DESCRIPTION
**Pull request by Pratyush**
## Description

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (master)
- [ ] Linked to GitHub issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer

### CODE REVIEW

**Review by mmagician — CHANGES_REQUESTED** [review]
Change log entry missing

**Comment by mmagician** [comment]
We were using a mix of parameters and config for naming, so I suppose that consistency is good. We should do the same for other repos.

**Comment by weikengchen** [comment]
I added a CHANGELOG entry. See if it is good:

```
- [\#545](https://github.com/arkworks-rs/algebra/pull/545) (`ark-ec`, `ark-ff`) Rename all `*Parameters` or `*Params` to `*Config`, including `SWUParams`, `WBParams`, `Bls12Parameters`, `G1Parameters`, `G2Parameters`, `BnParameters`, `BW6Parameters`, `MNT4Parameters`, `MNT6Parameters`, and `GLVParameters`. 
```

**Review by weikengchen — APPROVED** [review]
See mmagician's comment about the CHANGELOG

File: CHANGELOG.md [hunk 1]
```diff
@@ -141,6 +138,19 @@
...
+    - `Bls12Parameters` → `Bls12Config`
+    - `G1Parameters` → `G1Config`
+    - `G2Parameters` → `G2Config`
+    - `BnParameters` → `BnConfig`
+    - `BW6Parameters` → `BW6Config`
+    - `MNT4Parameters` → `MNT4Config`
+    - `MNT6Parameters` → `MNT6Config`
+    - `GLVParameters` → `GLVConfig`
+
+- [\#545](https://github.com/arkworks-rs/algebra/pull/545) (`ark-ec`) Rename all `*Parameters` to `*Config` for consistency.
```
Comment (Pratyush) [inline CHANGELOG.md:153]:
```suggestion
```

**Comment by weikengchen** [comment]
@Pratyush I also added a CHANGELOG entry. You can pick on to reconcile.

**Comment by Pratyush** [comment]
I picked yours, looks better.

**Comment by Pratyush** [comment]
@mmagician Changelog entry added.

**Review by mmagician — APPROVED** [review]

**Pull request merged**